File: 8_kyu/Count_by_X.go
package kata

// Makes array {x,2*x,3*x,...,n*x}.
func CountBy(x, n int) []int {
  result := make([]int,n)
  accu := x
  for i := range result {
    result[i] = accu
    accu += x
  }
  return result
}

File: 8_kyu/Count_by_X.c
// Fills the array result with 1*x, 2*x, 3*x ,, n*x.
void count_by(const unsigned x, const unsigned n, unsigned result[n]) {
  unsigned accu = x;
  for( unsigned it = 0; it < n; ++it, accu += x )
    result[it] = accu;
}
